### ISSUE TITLE
add more options to customize numbering style

### ISSUE THREAD

**Issue by alpianon** [lierdakil/pandoc-crossref#284]
When writing complex documents sometimes it could be useful to set complex custom numbering schemes as one can normally do in WYSIWYG word-processors, for instance:

```markdown
---
sectionsDepth: -1
numberSections: true
secHeaderDelim: ".) "
secLevelLabels:
  - alpha A
  - arabic
  - roman_lower
---

# First Section

text

## Subsection

other text

### Subsubsection

text text text
```
which should be rendered in this way:

># A) First Section
>
>text
>
>## A.1) Subsection
>
>other text
>
>### A.1.i) Subsubsection
>
>text text text

@lierdakil what do you think?

**Comment by kappapiana** [comment]
That would indeed be supercool.

**Comment by lierdakil** [comment]
This could be useful, yes. Unfortunately, implementation would be a little tricky. I'll see what I can do, but I can't promise I'll find the time to do this in the near future.

**Comment by lierdakil** [comment]
Actually, scratch that. I think I've figured out a way to do this a lot easier than I thought at first. The prototype is available from [actions](https://github.com/lierdakil/pandoc-crossref/actions/runs/377511040) or on [BinTray](https://bintray.com/lierdakil/pandoc-crossref/master/2cd1d9abb9b4c8e143ed0560dfab427480d8c5f8#files). An example is available [here](https://raw.githubusercontent.com/lierdakil/pandoc-crossref/master/test/m2m/secLevelLabels/input.md). Feel free to test and let me know if it needs more work.

**Comment by alpianon** [comment]
Great! You're one of the fastest maintainers I've ever seen :)

I'm trying it right now, I've just found a little glitch:

If someone by mistake sets both secLabels and secLevelLabels, this happens:

source:

```markdown
---
numberSections: true
sectionsDepth: -1
secLabels: roman
secLevelLabels:
  - alpha A
  - arabic
  - lowercase roman
---

# First Section

text

## Subsection

other text

### Subsubsection

text text text
```

html target:

> # AI First Section
> 
> text
> 
> ## AI.I Subsection
> 
> other text
> 
> ### AI.I.iI Subsubsection
> 
>
>text text text

I think that setting secLevelLabels should exclude secLabels, and if one by mistake sets both, a warning should be displayed in stderr, what do you think?

**Comment by lierdakil** [comment]
@alpianon, yep, that's a bug all right. The intention was for `secLevelLabels` to take precedence over `secLabels`, i.e. ignore the latter if the former is set. But I've messed up apparently. This should be fixed in 77f357590250326d8bc3d9ca777c86840a3396e1.

Displaying a warning is a bit tricky, because `secLabels` is set to `arabic` by default, so it's not exactly trivial to distinguish the case where it's explicitly set to `arabic` and the case when it's not set at all. Also, there's a question of where to draw the line exactly, i.e. do we display a warning when `secLabels` is set via `~/.pandoc-crossref/config.yaml`, or via `~/.pandoc-crossref/config-$fmt.yaml`, or from `crossrefYaml`? I've opted instead for the "mention it in the docs" route to avoid overthinking this.
